src/libraries/System.Collections.Immutable/src/Validation/Requires.cs

```diff
@@ -140,7 +140,7 @@ public static void FailObjectDisposed<TDisposed>(TDisposed disposed)
         {
             // separating out this throwing helps with inlining of the caller, especially
             // due to the retrieval of the type's name
-            throw new ObjectDisposedException(disposed!.GetType().FullName);
+            ObjectDisposedException.Throw(disposed!);
```

Comment:
This library builds for TFMs other than .NET 7.  You'll need to revert changes to code in any such libraries (it could be ifdef'd, but I don't think it's worth doing so).